bigquery/v2/query/client.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/bigquery/v2/apiv2/bigquerypb"
+	"cloud.google.com/go/bigquery/v2/apiv2_client"
+	"cloud.google.com/go/internal/uid"
+	"github.com/googleapis/gax-go/v2"
+	"google.golang.org/api/option"
+)
+
+// Client is a client for running queries in BigQuery.
+type Client struct {
+	c         *apiv2_client.Client
+	projectID string
+}
+
+// NewClient creates a new query client.
+func NewClient(ctx context.Context, projectID string, opts ...option.ClientOption) (*Client, error) {
+	qc := &Client{
+		projectID: projectID,
+	}
+	for _, opt := range opts {
+		if cOpt, ok := opt.(*customClientOption); ok {
+			cOpt.ApplyCustomClientOpt(qc)
+		}
+	}
+	if qc.c == nil {
+		c, err := apiv2_client.NewClient(ctx, opts...)
+		if err != nil {
+			return nil, fmt.Errorf("failed to setup bigquery client: %w", err)
+		}
+		qc.c = c
+	}
+	return qc, nil
+}
+
+// StartQuery runs a query and returns a QueryJob handle.
+func (c *Client) StartQuery(ctx context.Context, req *bigquerypb.PostQueryRequest, opts ...gax.CallOption) (*Query, error) {
+	if req.QueryRequest.RequestId == "" {
+		req.QueryRequest.RequestId = uid.NewSpace("request", nil).New()
+	}
+
+	res, err := c.c.Query(ctx, req, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run query: %w", err)
+	}
+
+	return newQueryJobFromQueryResponse(ctx, c, res)
+}
+
+// StartQueryJob from a bigquerypb.Job definition. Should have job.Configuration.Query filled out.
+func (c *Client) StartQueryJob(ctx context.Context, job *bigquerypb.Job, opts ...gax.CallOption) (*Query, error) {
+	qconfig := job.Configuration.Query
+	if qconfig == nil {
+		return nil, fmt.Errorf("job is not a query")
+	}
+
+	if job.Id == "" {
```

Comment:
Do you mean to construct the JobReference instead?  Id is output-only

Reply:
duh, wrong field 🤦